julia-ts-mode.el

```diff
@@ -270,24 +282,24 @@ Otherwise, the indentation is:
 
    :language 'julia
    :feature 'operator
-   `((adjoint_expression "'" @font-lock-type-face)
-     (let_binding (operator) @font-lock-type-face)
-     ((for_binding (operator) @font-lock-type-face)
-      (:match "^\\[=∈\\]$" @font-lock-type-face))
-     (arrow_function_expression "->" @font-lock-type-face)
-     (operator) @font-lock-type-face
-     (splat_expression "..." @font-lock-type-face)
-     (ternary_expression ["?" ":"] @font-lock-type-face)
-     (["." "::"] @font-lock-type-face))
+   `((adjoint_expression "'" @font-lock-keyword-face)
```

Comment:
Not blocking merge, but note that several of these should be either of `font-lock-operator-face` or `font-lock-misc-punctuation-face`.

Reply:
Is there a generic guideline on when to use which face? Frankly, I have difficulty deciding whether, for example, the `in` operator in a `for` loop head is an operator or a keyword. Likewise, the adjoint operator, or the splat operator, do these fall into the punctuation or into the operator category?

Would you give a suggestion on this list, where you would put each of these?

Reply:
I pushed a commit (d693c6b35d3aed986b2700a3b5f910de12d6c53c) that puts each of these where I think it should go, but I'm certainly open to disagreement on the choices I made there. The only source I'm aware of for guidelines is [the elisp manual](https://www.gnu.org/software/emacs/manual/html_node/elisp/Faces-for-Font-Lock.html) which is often too concise to be helpful. Other than that, I try to look at what some of the major modes included with python do (e.g. python-mode, ruby-mode).

Reply:
Yes, I saw that, thank you! Some choices seem debatable, but if I squint, I can see the logic behind them. At first glance, for example, it seems odd that the `=` is an operator in a `let_binding`, but a keyword in a `for_binding`. On closer inspection, though, I can understand that it is a regular assignment operator in the first case, and something quite different in the second.

Generally speaking, I'm okay with them, especially considering that none of these appear on the default font-lock level.